File: src/main/java/com/pauljoda/modularsystems/core/ModularSystems.java
package com.pauljoda.modularsystems.core;

import com.pauljoda.modularsystems.core.api.nei.INEICallback;
import com.pauljoda.modularsystems.core.commands.AddBannedBlock;
import com.pauljoda.modularsystems.core.commands.AddFluidFuel;
<<<<<<< HEAD
import com.pauljoda.modularsystems.core.commands.OpenValueConfig;
=======
import com.pauljoda.modularsystems.core.functions.VersionCheckerFunction;
>>>>>>> origin/master
import com.pauljoda.modularsystems.core.lib.Reference;
import com.pauljoda.modularsystems.core.managers.BlockManager;
import com.pauljoda.modularsystems.core.managers.GuiManager;
import com.pauljoda.modularsystems.core.managers.RecipeManager;
import com.pauljoda.modularsystems.core.proxy.CommonProxy;
import com.pauljoda.modularsystems.core.registries.*;
import com.pauljoda.modularsystems.core.managers.PacketManager;
import cpw.mods.fml.common.Mod;
import cpw.mods.fml.common.SidedProxy;
import cpw.mods.fml.common.event.FMLInitializationEvent;
import cpw.mods.fml.common.event.FMLPostInitializationEvent;
import cpw.mods.fml.common.event.FMLPreInitializationEvent;
import cpw.mods.fml.common.event.FMLServerStartingEvent;
import cpw.mods.fml.common.network.NetworkRegistry;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.command.ICommandManager;
import net.minecraft.command.ServerCommandManager;
import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.item.Item;
import net.minecraft.server.MinecraftServer;
import net.minecraftforge.common.MinecraftForge;

import java.io.File;

@Mod(name = Reference.MOD_NAME, modid = Reference.MOD_ID, version = Reference.VERSION, dependencies = Reference.DEPENDENCIES)
public class ModularSystems {
    @Mod.Instance(Reference.MOD_ID)
    public static ModularSystems instance;

    @SidedProxy( clientSide="com.pauljoda.modularsystems.core.proxy.ClientProxy", serverSide="com.pauljoda.modularsystems.core.proxy.CommonProxy")
    public static CommonProxy proxy;

    public static INEICallback nei;

    public static String configFolderLocation;

    //Creates the Creative Tab
    public static CreativeTabs tabModularSystems = new CreativeTabs("tabModularSystems") {
        @Override
        @SideOnly(Side.CLIENT)
        public Item getTabIconItem() {
            return Item.getItemFromBlock(BlockManager.furnaceCore);
        }
    };

    @Mod.EventHandler
    public void preInit(FMLPreInitializationEvent event) {
        BlockManager.init();

        MinecraftForge.EVENT_BUS.register(FurnaceBannedBlocks.INSTANCE);
        MinecraftForge.EVENT_BUS.register(BlockValueRegistry.INSTANCE);
        MinecraftForge.EVENT_BUS.register(FluidFuelValues.INSTANCE);
        NetworkRegistry.INSTANCE.registerGuiHandler(this, new GuiManager());

        configFolderLocation = event.getModConfigurationDirectory().getAbsolutePath() + File.separator + "Modular-Systems";
        GeneralConfigRegistry.init(configFolderLocation);

        proxy.preInit();
    }

    @Mod.EventHandler
    public void init(FMLInitializationEvent event) {
        FurnaceBannedBlocks.INSTANCE.init();
        BlockValueRegistry.INSTANCE.init();
        FluidFuelValues.INSTANCE.init();
        CrusherRecipeRegistry.INSTANCE.init();

        RecipeManager.init();
        PacketManager.initPackets();

        proxy.init();

        VersionCheckerFunction.execute();
    }

    @Mod.EventHandler
    public void postInit(FMLPostInitializationEvent event) { }

    @Mod.EventHandler
    public void serverLoad(FMLServerStartingEvent event) {
        MinecraftServer server = MinecraftServer.getServer();
        ICommandManager command = server.getCommandManager();
        ServerCommandManager manager = (ServerCommandManager) command;
        manager.registerCommand(new AddBannedBlock());
        manager.registerCommand(new AddFluidFuel());
        manager.registerCommand(new OpenValueConfig());
    }
}

File: src/main/scala/com/teambr/modularsystems/core/ModularSystems.scala
package com.teambr.modularsystems.core

import java.io.File

import com.teambr.modularsystems.core.achievement.ModAchievements
import com.teambr.modularsystems.core.commands.OpenValueConfig
import com.teambr.modularsystems.core.common.CommonProxy
import com.teambr.modularsystems.core.lib.Reference
import com.teambr.modularsystems.core.managers._
import com.teambr.modularsystems.core.network.PacketManager
import com.teambr.modularsystems.core.registries._
import com.teambr.modularsystems.storage.event.ToolTipEvent
import net.minecraft.command.ServerCommandManager
import net.minecraft.creativetab.CreativeTabs
import net.minecraft.init.Blocks
import net.minecraft.item.Item
import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.fml.common.Mod.EventHandler
import net.minecraftforge.fml.common.event.{FMLInitializationEvent, FMLPostInitializationEvent, FMLPreInitializationEvent, FMLServerStartingEvent}
import net.minecraftforge.fml.common.network.NetworkRegistry
import net.minecraftforge.fml.common.{Mod, SidedProxy}
import net.minecraftforge.fml.relauncher.{Side, SideOnly}
import org.apache.logging.log4j.LogManager

/**
  * This file was created for the Modular-Systems
  * API. The source is available at:
  * https://github.com/TeamCoS/Modular-Systems
  *
  * Bookshelf if licensed under the is licensed under a
  * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
  * http://creativecommons.org/licenses/by-nc-sa/4.0/
  *
  * @author Paul Davis <pauljoda>
  * @since August 02, 2015
  */
@Mod(modid = Reference.MOD_ID, name = Reference.MOD_NAME, version = Reference.VERSION,
    dependencies = Reference.DEPENDENCIES, modLanguage = "scala")
object ModularSystems {
    // In Java, allows us to get the instance
    def getInstance = this

    //The logger. For logging
    final val logger = LogManager.getLogger(Reference.MOD_NAME)

    var configFolderLocation : String = ""

    @SidedProxy(clientSide = "com.teambr.modularsystems.core.client.ClientProxy",
        serverSide = "com.teambr.modularsystems.core.common.CommonProxy")
    var proxy : CommonProxy = null

    var tabModularSystems : CreativeTabs = new CreativeTabs("tabModularSystems") {
        @SideOnly(Side.CLIENT)
        override def getTabIconItem: Item = Item.getItemFromBlock(Blocks.FURNACE)
    }

    @EventHandler def preInit(event : FMLPreInitializationEvent) = {
        configFolderLocation = event.getModConfigurationDirectory.getAbsolutePath + File.separator + "Modular-Systems"
        ConfigRegistry.preInit()
        BlockManager.preInit()
        ItemManager.preInit()
        FurnaceBannedBlocks.init()
        BlockValueRegistry.INSTANCE.init()
        FluidFuelValues.init()
        proxy.preInit()
        CraftingManager.init()
        ModAchievements.start()
        NetworkRegistry.INSTANCE.registerGuiHandler(this, new GuiManager)

        MinecraftForge.EVENT_BUS.register(BlockValueRegistry.INSTANCE)
        MinecraftForge.EVENT_BUS.register(EventManager.INSTANCE)
        MinecraftForge.EVENT_BUS.register(new ToolTipEvent)
    }

    @EventHandler def init(event : FMLInitializationEvent) =  {
        ItemManager.init() // Must be before proxy registration
        CrusherRecipeRegistry.init() //Must be after ItemManager.init
        PacketManager.initPackets()
        proxy.init()
        //FMLCommonHandler.instance().bus().register(ModAchievements)
        MinecraftForge.EVENT_BUS.register(ModAchievements)
    }

    @EventHandler def postInit(event : FMLPostInitializationEvent) = {
        proxy.postInit()
    }


    @Mod.EventHandler
    def serverLoad(event : FMLServerStartingEvent) {
        event.getServer.getCommandManager.asInstanceOf[ServerCommandManager].registerCommand(new OpenValueConfig())
    }
}
